### PR TITLE
Basic QML Sidebar placeholder

### PR DESCRIPTION
**Pull request by Holzhaus**
Follow-up to #4034.

This now exposes the SidebarModel to QML and adds an actual treeview to the QML skin.
Top level items are displayed, including icons.

For now, those items are not clickable (nothing happens) and the lazy loading mechanic (e.g. browse feature) is not working. For stuff like that to work, some more serious refactoring is needed. The same goes for child item icons.

Unfortunately, Qt Quick Control 2 does not provide a TreeView (they made that a [premium feature](https://marketplace.qt.io/products/treeview)), so this still uses the old Qt Quick Controls 1 TreeView component. We may evaluate [existing solutions](https://github.com/diracsbracket/qmltreeview2) or [roll our own](https://www.qtdeveloperdays.com/sites/default/files/north-america/QtQuickTreeView.pdf) in the future.

### CODE REVIEW

**Comment by Be-ing** [comment]
Qt does have a TreeView for QQC2 now, but obnoxiously it is in the ["Qt marketplace"](https://marketplace.qt.io/products/treeview) instead of packaged with Qt. It is licensed GPL to pressure companies using Qt as LGPL to pay for proprietary licenses.

**Comment by Be-ing** [comment]
There is some discussion about packaging difficulties with the Qt Marketplace thing on this [bug report](https://bugreports.qt.io/browse/QTBUG-61630). Looking at the [Git repository](https://code.qt.io/cgit/qt-extensions/qttreeview.git), it seems they recently added or are currently working on Qt6 support with CMake.

**Comment by Be-ing** [comment]
Kirigami Addons also has a [TreeView](https://api.kde.org/frameworks-api/frameworks-apidocs/frameworks/kirigami-addons/html/classorg_1_1kde_1_1kirigamiaddons_1_1treeview_1_1TreeListView.html).

**Comment by Holzhaus** [comment]
Hah, I linked the marketplace tree view above, but I didn't notice the "Also available:GPLv3" and thought it was premium only.

**Comment by Holzhaus** [comment]
> Kirigami Addons also has a [TreeView](https://api.kde.org/frameworks-api/frameworks-apidocs/frameworks/kirigami-addons/html/classorg_1_1kde_1_1kirigamiaddons_1_1treeview_1_1TreeListView.html).

I'd like to avoid pulling in all kinds of KDE dependencies. I'll check the marketplace tree view first.

**Comment by Be-ing** [comment]
Kirigami does not depend on other KDE Frameworks libraries.

**Comment by uklotzde** [comment]
An MVP with limited functionality is fine. Let's first explore how far we can get with the available components before we pull in new libraries. Maybe we discover a more appropriate solution for the side bar functionality along the way.

I like the lightweight approach of first mimicking the status quo for reusing as much of the existing functionality as possible and to determine any functional or conceptual gaps.

**Comment by Holzhaus** [comment]
> Let's first explore how far we can get with the available components before we pull in new libraries.

I agree. The issue here is that Qt Quick Controls 1 is deprecated and removed in Qt6. So if we want to get it working with Qt6, we need to get rid of the Qt Quick Controls 1 TreeView first.

I already tried to get the Qt Quick 2 tree view working, but there is no cmakelists for Qt5, just Qt 6. I'm looking into it.

File: src/skin/qml/qmllibraryproxy.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class QmlLibraryProxy : public QObject {
+    Q_OBJECT
+  public:
+    explicit QmlLibraryProxy(Library* pLibrary, QObject* parent = nullptr);
+
+    Q_INVOKABLE QAbstractItemModel* getSidebarModel();
+
+  private:
+    Library* m_pLibrary;
```
Comment (uklotzde) [inline src/skin/qml/qmllibraryproxy.h:22]: How about using `QPointer` or `mixxx::SafeQPointer` to prevent dangling pointers?

**Review by uklotzde — APPROVED** [review]
Considered mergeable.

**Review by Swiftb0y — APPROVED** [review]
In the long run, we should think about a Sidebar-less library as already mentioned on zulip but I consider this approach fine as well for now.

**Review by Swiftb0y — APPROVED** [review]
LGTM, I'll go ahead and merge

**Pull request merged**